watcher_fsevents.go

```diff
@@ -189,9 +187,6 @@ func newWatcher(c chan<- EventInfo) watcher {
 }
 
 func (fse *fsevents) watch(path string, event Event, isrec int32) (err error) {
-	if path, err = canonical(path); err != nil {
```

Comment:
I don't think we can remove this yet, until the functionality is moved to tree.

Morever I think it'd need to stay here because of `fse.watches` - if we put a watch on a dir and a symlink to it, `fse.watches` must not hold two watches but only one.

Reply:
Isn't it already in the tree:
https://github.com/rjeczalik/notify/blob/master/tree_recursive.go#L164

Cleanpath subsequently calls canonical on the path and no other watcher implementation uses canonical.

Reply:
It is, but `fse.watches` maintains its own path -> watches mapping, if we get rid of canonical, we may have duplicated watches.

Reply:
I still don't understand it: The paths in `fse.watches` that were previously passed through canonical are those given as arguments to the exported (Recursive-)Re-/Watch functions, which are called from the tree, where the paths were already canonicalized in the tree - how else can duplicated paths enter `fse.watches`?

Other watchers maintains such a map of paths to structs as well, but don't canonicalize paths (e.g. readdcw in `r.m`, trigger in `pthLkp`).

Also the new tests fail when `canonical` is reintroduced, as it won't remove a non-existing path, because `canonical` stats the path and thus fails.

Reply:
> which are called from the tree, where the paths were already canonicalized in the tree

You're right, I take back my comment about duplicates - canonical is not needed here.

> Also the new tests fail when canonical is reintroduced, as it won't remove a non-existing path, because canonical stats the path and thus fails.

Good point.